scripts/dependencies/tools.ts

```diff
@@ -558,3 +558,26 @@ export class ECRCredHelper implements Dependency, GitHubDependency {
     return semver.rcompare(version1, version2);
   }
 }
+
+export class SpinShim implements Dependency {
+  name = 'containerd-shim-spin-v2';
+  githubOwner = 'deislabs';
+  githubRepo = 'containerd-wasm-shims';
+
+  async download(context: DownloadContext): Promise<void> {
+    const arch = context.isM1 ? 'aarch64' : 'x86_64';
+    const base = `https://github.com/${ this.githubOwner }/${ this.githubRepo }/releases/download/v${ context.versions.wasmShims }`;
+    const url = `${ base }/containerd-wasm-shims-v2-spin-linux-${ arch }.tar.gz`;
+    const destPath = path.join(context.resourcesDir, 'linux', 'internal', this.name);
+
+    await downloadTarGZ(url, destPath);
+  }
+
+  async getAvailableVersions(includePrerelease = true): Promise<string[]> {
+    return await getPublishedVersions(this.githubOwner, this.githubRepo, includePrerelease);
```

Comment:
Looking at https://github.com/deislabs/containerd-wasm-shims/releases/tag/v0.11.0 it doesn't seem to include `spin`? Do we need to do some filtering?

Hmm, but it previously downloaded fine in [the CI run](https://github.com/rancher-sandbox/rancher-desktop/actions/runs/7894659486/job/21545725993#step:9:60), so maybe they took it down?

Reply:
They seem to have screwed it up. 😢 

See also https://github.com/deislabs/containerd-wasm-shims/issues/211#issuecomment-1943907264

We can downgrade to 0.10.0 for now.